### PR TITLE
add vim plugins

### PR DESCRIPTION
**Pull request by RonanMacF**
all plugins added via vim-plugins/update.py

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 123920` at 33c6b6fc run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/9656dbb89014731398d48b590ae55e68">vimPlugins.vim-colorschemes</a></li>
  </ul>
</details>
<details>
  <summary>8 packages built successfully:</summary>
  <ul>
    <li>spacevim</li>
    <li>vimPlugins.conflict-marker-vim</li>
    <li>vimPlugins.hiPairs</li>
    <li>vimPlugins.nvim-dap-ui</li>
    <li>vimPlugins.nvim-treesitter-pyfold</li>
    <li>vimPlugins.pears-nvim</li>
    <li>vimPlugins.scrollbar-nvim</li>
    <li>vimPlugins.telescope-dap-nvim</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/misc/vim-plugins/build-vim-plugin.nix:28:14:
```
   |
28 |       inherit unpackPhase configurePhase buildPhase addonInfo preInstall postInstall;
   |              ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/misc/vim-plugins/build-vim-plugin.nix:28:14:
```
   |
28 |       inherit unpackPhase configurePhase buildPhase addonInfo preInstall postInstall;
   |              ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/misc/vim-plugins/generated.nix [hunk 1]
```diff
@@ -5661,7 +5745,7 @@ final: prev:
       owner = "flazz";
       repo = "vim-colorschemes";
       rev = "fd8f122cef604330c96a6a6e434682dbdfb878c9";
-      sha256 = "1cg8q7w0vgl73aw1b9zz0zh5vw5d2pm8pm54fhfzva4azg56f416";
+      sha256 = "0kpsf6j20fgblc8vhqn7ymr52v2d1h52vc7rbxmxfwdm80nvv3g5";
```
Comment (jonringer) [inline pkgs/misc/vim-plugins/generated.nix:5748]: this doesn't seem to be right
```
hash mismatch in fixed-output derivation '/nix/store/2r0hrjgnhb8nyf5zh0gn8xwyqxazrg9f-source':
  wanted: sha256:0kpsf6j20fgblc8vhqn7ymr52v2d1h52vc7rbxmxfwdm80nvv3g5
  got:    sha256:1cg8q7w0vgl73aw1b9zz0zh5vw5d2pm8pm54fhfzva4azg56f416
```

Comment (RonanMacF) [reply pkgs/misc/vim-plugins/generated.nix]: not sure how that occurred, anyways have kept it as 1cg8q7w0vgl73aw1b9zz0zh5vw5d2pm8pm54fhfzva4azg56f416

Comment (SuperSandro2000) [reply pkgs/misc/vim-plugins/generated.nix]: If ofborg is telling it is wrong it needs to be updated. Also please resolve the merge conflict.

Comment (RonanMacF) [reply pkgs/misc/vim-plugins/generated.nix:5760]: rebased and ran `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"` which succeeded

Comment (SuperSandro2000) [reply pkgs/misc/vim-plugins/generated.nix:5760]:
```suggestion
      sha256 = "sha256-JhBnyvuKqP0ddKTUi+oVrfBd4Af/pxW4Goe+DfjB6LE=";
```

**Pull request merged**